cql3/attributes.cc

```diff
@@ -107,6 +115,19 @@ db::timeout_clock::duration attributes::get_timeout(const query_options& options
     return std::chrono::duration_cast<db::timeout_clock::duration>(std::chrono::nanoseconds(duration.nanoseconds));
 }
 
+qos::service_level_options attributes::get_service_level(qos::service_level_controller& sl_controller, const query_options& options) const {
+    cql3::raw_value service_level = expr::evaluate(*_service_level, options);
+    if (service_level.is_null()) {
+        throw exceptions::invalid_request_exception("Service level value cannot be null");
```

Comment:
We can treat NULL the same as the WITH SERVICE LEVEL clause not existing (fall back to role-based selection).

Reply:
Now, as I removed support for bind markers and service level name is parsed as sstring(not as expression), null value is not an option.